src/main/java/athleticli/dietgoal/DietGoal.java

```diff
@@ -25,10 +25,23 @@ public String getNutrients() {
         return nutrients;
     }
 
+    public void setNutrients(String nutrients) {
+        this.nutrients = nutrients;
+    }
+
     public int getTargetValue() {
         return targetValue;
     }
 
+    public void setTargetValue(int targetValue) {
+        this.targetValue = targetValue;
+        if (!isGoalAchieved && currentValue >= targetValue) {
+            setIsGoalAchieved(true);
+        } else if (isGoalAchieved && currentValue < targetValue) {
+            setIsGoalAchieved(false);
+        }
```

Comment:
Should we simplify the logic and always set depending on the currentValue?
```suggestion
        if (currentValue >= targetValue) {
            setIsGoalAchieved(true);
        } else {
            setIsGoalAchieved(false);
        }
```

Reply:
I think this is a good suggestion: it won't hurt to re-assign `isGoalAchieved` to its current value, and it in fact can simplify the code logic.

I think changing the code to the following will simplify the code, but I'm not sure whether doing so will reduce readability:

```suggestion
        setIsGoalAchieved(currentValue >= targetValue);
```